src/System.Net.Http/tests/StressTests/HttpStress/Program.cs

```diff
@@ -0,0 +1,377 @@
+                {
+                    using (RSA rsa = RSA.Create())
+                    {
+                        var certReq = new CertificateRequest($"CN={LocalhostName}", rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+                        certReq.CertificateExtensions.Add(new X509BasicConstraintsExtension(false, false, 0, false));
+                        certReq.CertificateExtensions.Add(new X509EnhancedKeyUsageExtension(new OidCollection { new Oid("1.3.6.1.5.5.7.3.1") }, false));
+                        certReq.CertificateExtensions.Add(new X509KeyUsageExtension(X509KeyUsageFlags.DigitalSignature, false));
+                        X509Certificate2 cert = certReq.CreateSelfSigned(DateTimeOffset.UtcNow.AddMonths(-1), DateTimeOffset.UtcNow.AddMonths(1));
+                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        {
+                            cert = new X509Certificate2(cert.Export(X509ContentType.Pfx));
+                        }
+                        listenOptions.UseHttps(cert);
+                    }
+                });
+            })
+
+            // Output only warnings and errors from Kestrel
+            .ConfigureLogging(log => log.AddFilter("Microsoft.AspNetCore", level => level >= LogLevel.Warning))
+            
+            // Set up how each request should be handled by the server.
+            .Configure(app =>
+            {
+                app.UseRouting();
+                app.UseEndpoints(endpoints =>
+                {
+                    endpoints.MapGet("/", async context =>
+                    {
+                        // Get requests just send back the requested content.
+                        await context.Response.WriteAsync(content);
+                    });
+                    endpoints.MapPost("/", async context =>
+                    {
+                        // Post echos back the requested content, first buffering it all server-side, then sending it all back.
+                        var s = new MemoryStream();
+                        await context.Request.Body.CopyToAsync(s);
+                        await context.Response.StartAsync();
+                        s.Position = 0;
+                        await s.CopyToAsync(context.Response.Body);
+                    });
+                    endpoints.MapPost("/duplex", async context =>
+                    {
+                        // Echos back the requested content in a full duplex manner.
+                        await context.Request.Body.CopyToAsync(context.Response.Body);
+                    });
+                });
+            })
+            .Build()
+            .Start();
+
+        // Configure the client.
+        Console.WriteLine($"Starting {ConcurrentRequests} client workers.");
+        using (var handler = new SocketsHttpHandler() { SslOptions = new SslClientAuthenticationOptions { RemoteCertificateValidationCallback = delegate { return true; } } })
```

Comment:
just wondering...why create `SocketsHttpHandler` directly?  Why not just create `HttpClientHandler`?

Reply:
Simply because
a) This is primarily about testing the HTTP/2 implementation in SocketsHttpHandler.
b) I expect we'll want to explore configuring other settings on it.
It's not a big deal, though.